HumanDynamicsEstimationLibrary/utils/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+find_package(OsqpEigen 0.4.0 REQUIRED)
+find_package(YARP 3.2 REQUIRED)
+find_package(iDynTree REQUIRED)
+
+# Include AddComponent cmake module
+include(AddComponent)
+
+add_component(NAME utils
+              SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/src/iDynTreeUtils.cpp
+              PUBLIC_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/hde/utils/iDynTreeUtils.hpp
+              PUBLIC_LINK_LIBRARIES YARP::YARP_OS
+                                    YARP::YARP_dev
+                                    YARP::YARP_init
+                                    iDynTree::idyntree-model
+                                    iDynTree::idyntree-inverse-kinematics
+                                    iDynTree::idyntree-high-level
+              PRIVATE_LINK_LIBRARIES OsqpEigen::OsqpEigen
+                                     osqp::osqp
```

Comment:
Just checking, are all of these necessary?

Reply:
I will double check. Anyway, dependency on yarp at this stage should be just related to logging messages. Indeed, we can remove it (in a separate PR), moving for example to spdlog as done in [blf](https://github.com/dic-iit/bipedal-locomotion-framework/pull/225)

Reply:
From the `iDynTreeUtils.h` header, it seems that you just need `iDynTree::idyntree-core`. In the cpp instead (so the private ones) it seems you need just `YARP::YARP_OS`. You could remove all the other ones, including osqp.

Reply:
Thanks, I have cleaned the includes with https://github.com/robotology/human-dynamics-estimation/pull/246